pkg/comp-functions/functions/common/instance_namespace.go

```diff
@@ -105,7 +105,7 @@ func createInstanceNamespace(serviceName, compName, claimNamespace, instanceName
 		ObjectMeta: metav1.ObjectMeta{
 			Name: instanceNamespace,
 			Labels: map[string]string{
-				"appcat.vshn.io/servicename":     serviceName + "-standalone",
+				"appcat.vshn.io/servicename":     serviceName + "-" + svc.Config.Data["mode"],
```

Comment:
Can You do some validation before? If You try to access key in map that doesn't exist, then it's going to be a nil pointer reference and ugly stack trace in logs at least, worst - restart of container

Reply:
there will be no nil pointer but empty string. I checked a random composition and I didn't see "mode" value in config. It will end up as "<servicename>-". I would change to default if string is missing thus "-standalone"

Reply:
It's going to be set in any case in the component. Not sure if we should really check for that. And afaik it should not return a null pointer but an empty string as a result?

Reply:
I'd go with a default as well.

something like this:
```go
mode := "standalone"
if _, exists := svc.Config.Data["mode"]; exists {
  mode = svc.Config.Data["mode"]
}
```

Reply:
Thanks for the suggestions. I've implemented it accordingly